nucypher/blockchain/eth/actors.py

```diff
@@ -1493,76 +1493,75 @@ def was_this_evidence_evaluated(self, evidence) -> dict:
         return receipt
 
 
-class StakeHolder(Staker):
+class Wallet:
+    """
+    Account management abstraction on top of blockchain providers and external signers
+    """
+    class UnknownAccount(KeyError):
+        pass
 
-    banner = STAKEHOLDER_BANNER
+    def __init__(self,
+                 client_addresses: set = None,
+                 provider_uri: str = None,
+                 signer=None):
 
-    class StakingWallet:
+        self.__client_accounts = list()
+        self.__transacting_powers = dict()
 
-        class UnknownAccount(KeyError):
-            pass
+        # Blockchain
+        self.blockchain = BlockchainInterfaceFactory.get_interface(provider_uri)
+        self.__signer = signer
+
+        self.__get_accounts()
+        if client_addresses:
+            self.__client_accounts.extend([a for a in client_addresses if a not in self.__client_accounts])
+
+    @validate_checksum_address
+    def __contains__(self, checksum_address: str) -> bool:
+        return bool(checksum_address in self.accounts)
 
-        def __init__(self,
-                     registry: BaseContractRegistry,
-                     client_addresses: set = None,
-                     signer=None):
+    @property
+    def active_account(self) -> str:
+        return self.blockchain.transacting_power.account
```

Comment:
Perhaps the codebase has finally gotten to a place where we can remove the notion of an "active account".  Is it fulfilling an important function from your view, @cygnusv ?